models/optimism_attestationstation/optimism/optimism_attestationstation_optimism_events.sql

```diff
@@ -11,31 +11,45 @@
                                 \'["chuxin"]\') }}'
     )
 }}
-select 
-    date_trunc('day', evt_block_time) as block_date
-    ,evt_tx_hash as tx_hash
-    ,evt_block_number as block_number
-    ,evt_block_time as block_time
-    ,evt_index
-    ,about as recipient
-    ,creator as issuer
-    ,contract_address
-    ,key as key_raw
-    ,decode(
-        unhex(
-          if (
-            substring(key, 1, 6) in ("0xab7e", "0x9e43"),
-            hex(key),
-            substring(key, 3)
-          )
-        ),
-        "utf8"
-      ) as key
-    ,val as val_raw
-    ,split(unhex(substring(val, 3)), ",") as val
-from {{source('attestationstation_optimism','AttestationStation_evt_AttestationCreated')}}
-where 
-    true
-    {% if is_incremental() %}
-    and evt_block_time >= date_trunc('day', now() - interval '1 week')
-    {% endif %}
+SELECT
+  *
+  , concat_ws(', ', val) AS val_string
+
+  FROM (
+    select 
+        date_trunc('day', evt_block_time) as block_date
+        ,evt_tx_hash as tx_hash
+        ,evt_block_number as block_number
+        ,evt_block_time as block_time
+        ,evt_index
+        ,about as recipient
+        ,creator as issuer
+        ,contract_address
+        ,key as key_raw
+        ,
+        REGEXP_REPLACE( --Replace invisible characters
+            decode(
+                unhex(
+                  if (
+                    substring(key, 1, 6) in ("0xab7e", "0x9e43"), --Handle for Clique
+                    hex(key),
+                    substring(key, 3)
+                  )
+                ),
+                "utf8"
+              ) 
+        , '[[:cntrl:]]', '')
+        as key
+        ,val as val_raw
+
+        ,split(unhex(substring(val, 3)), ",") as val
+
+        ,bytea2numeric(substring(val, 3)) AS val_byte2numeric
```

Comment:
we have a `v3` of this function -- does it make sense to use [this](https://github.com/duneanalytics/spellbook/blob/main/macros/public/bytea2numeric_v3.sql)?

Reply:
looks complicated, but if it's "better" then makes sense.

Reply:
Also cool to see a function example!